pkg/trait/builder.go

```diff
@@ -256,7 +256,10 @@ func (t *builderTrait) buildahTask(e *Environment) (*v1.ImageTask, error) {
 	}
 
 	var auth string
-	if e.Platform.Status.Build.Registry.Secret != "" {
+	if e.Platform.Status.Build.Registry.Insecure {
```

Comment:
I wonder how symmetrical it is or should be with what's done with Kaniko? For the later, we've decided to de-correlate the two options, so we may want to do the same for Buildah.

Reply:
Do you mean doing something like:

```if !strings.HasPrefix(image, e.Platform.Status.Build.Registry.Address) && e.Platform.Status.Build.Registry.Insecure  {
}
```

Such as it prioritizes it only if the registry for the image matches the registry address?

Reply:
Sorry, I meant for Kaniko, we are doing:

```go
if e.Platform.Status.Build.Registry.Secret != "" {

}
if e.Platform.Status.Build.Registry.Insecure {

}
```

While here we propose:

```go
if e.Platform.Status.Build.Registry.Insecure {

}
} else if e.Platform.Status.Build.Registry.Secret != "" {

}
```
I would find it logical to chose a strategy and apply the same for Kaniko and Buildah.

Reply:
@astefanutti oh, I got your point. That's a very good one and I agree with you that it should be streamlined so that it behaves the same on this strategy as well. IMHO, we could align to the way we are doing with Kaniko. What do you think? If you are fine with it, I will adjust it that way.

Reply:
I agree, let's align with what we are doing with Kaniko 👍🏼.